internal/testutils/provider.go

```diff
@@ -223,27 +225,40 @@ func ExpiryDeviceAuthHandler() ProviderHandler {
 
 // MockProviderInfoer is a mock that implements the ProviderInfoer interface.
 type MockProviderInfoer struct {
+	T *testing.T
+
 	Scopes    []string
 	Options   []oauth2.AuthCodeOption
 	Groups    []info.Group
 	GroupsErr bool
+
+	ExpectFailureToCastTokenScopes bool
+	ExpectMissingScopes            bool
 }
 
 // CheckTokenScopes checks if the token has the required scopes.
 func (p *MockProviderInfoer) CheckTokenScopes(token *oauth2.Token) error {
 	scopesStr, ok := token.Extra("scope").(string)
-	if !ok {
-		return fmt.Errorf("failed to cast token scopes to string: %v", token.Extra("scope"))
+	if p.ExpectFailureToCastTokenScopes {
```

Comment:
instead of the `if`/`else`, I would just assert on `require.Equal()` and match `ok` with it. This will allow the assertion to be a single line and no `else` (see below)